### ISSUE TITLE
Improved developer docs/undocumented non-python dependencies

### ISSUE THREAD

**Issue by mlittlej** [mbj4668/pyang#811]
There are a couple of non-python dependencies that developers need to install before they can get the unit-tests running successfully when they start working on pyang's code.

For me, working on a reasonably clean Debian (Bookworm) OS, they were/are:

* xsltproc
* xmllint (libxml2-utils)
* jing

There may be other "hidden" dependencies that I already had installed on my machine - I don't know.

It'd be helpful to have these dependencies documented somewhere up-front so that someone coming to pyang for the first time, or on a new machine, can pre-install everything before running the unit-tests. The unit-tests failing because a particular utility wasn't found in my $PATH was the only way for me to discover the problem(s) and the UTs didn't run especially quickly for me.

**Comment by mbj4668** [comment]
Good point!  As you can see in https://github.com/mbj4668/pyang/blob/master/.github/workflows/tests.yml, you have found the required dependencies.   But as you note there might be other dependencies like `grep`, `diff` etc that are typically installed already.
